### PR TITLE
cniserver: Fix HNS endpoints leak on Windows

### PR DESCRIPTION
**Pull request by alinbalutoiu**
Currently, if attaching HNS endpoint to the container fails,
the HNS endpoint does not get deleted. This happens because
the delete function receives the ID of the container instead
of the name of the HNS endpoint.

This also fixes another possible HNS endpoints leak that can
happen if running the ovs command fails.

Fixes issue: #390 

Signed-off-by: Alin Balutoiu <alinbalutoiu@gmail.com>

### CODE REVIEW

**Comment by alinbalutoiu** [comment]
cc: @rajatchopra @aserdean

File: go-controller/pkg/cni/helper_windows.go [hunk 1]
```diff
@@ -191,9 +196,10 @@ func (pr *PodRequest) ConfigureInterface(namespace string, podName string, macAd
 		fmt.Sprintf("-NlMtuBytes %d", mtu),
 	}
 
-	_, err = exec.Command("powershell", mtuArgs...).CombinedOutput()
-	if err != nil {
-		logrus.Warningf("Failed to set MTU on endpoint, reason: %q", err)
+	// Using errMTU in order to not delete the endpoint when setting MTU fails
```
Comment (aserdean) [inline go-controller/pkg/cni/helper_windows.go]: Shouldn't we delete the endpoint if setting the MTU fails?

**Comment by alinbalutoiu** [comment]
Synced with @aserdean offline.
PTAL @rajatchopra

**Comment by aserdean** [comment]
LGTM

**Pull request merged**